File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    static Queue<Integer>q=new LinkedList<Integer>();
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> l=new ArrayList<Integer>();
         traversal(root,0);
        while(!q.isEmpty())
            l.add(q.poll());
        return l;
    }
    public void traversal(TreeNode root, int level)
    {
        if (root==null)
            return ;
        if (level==q.size())
            q.add(root.val);
        traversal(root.right,level+1);
        traversal(root.left,level+1);
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        l=[]
        self.rightview(root,0,l)
        return l
    def rightview(self,root,level,l):
        if root==None:
            return
        if level==len(l):
            l.append(root.val)
        self.rightview(root.right,level+1,l)
        self.rightview(root.left,level+1,l)
        